services/order.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+@transaction.atomic
+def create_order(
+        tickets: list,
+        username: str,
+        date: Optional[str] = None
+) -> Order:
+    user = get_user_model().objects.get(username=username)
+    new_order = Order.objects.create(user=user)
+
+    if date:
+        new_order.created_at = date
+        new_order.save()
+
+    for ticket_data in tickets:
+        Ticket.objects.create(
+            row=ticket_data["row"],
+            seat=ticket_data["seat"],
+            movie_session_id=ticket_data["movie_session"],
+            order=new_order
+        )
+
+    return new_order
+
+
+def get_orders(username: Optional[str] = None) -> QuerySet:
+    if username:
+        return Order.objects.filter(user__username=username)
+    return Order.objects.all()

```

Comment:
Better to do like that, because when you add new params it will be simple to modify
```suggestion
    queryset = Order.objects.all()
    if username:
        queryset = queryset.filter(user__username=username)
    return queryset
```

Reply:
Thank you for all the remarks. Updated already.